app/Singletons/UserCoverPresets.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Singletons;
+
+use App\Models\User;
+use App\Models\UserCoverPreset;
+use App\Traits\Memoizes;
+use App\Transformers\UserCoverPresetTransformer;
+use Illuminate\Support\Collection;
+
+class UserCoverPresets
```

Comment:
wonder if it's worth calling `resetMemoized()` when the covers are updated in the controller?